tools/pythonpkg/src/pyrelation.cpp

```diff
@@ -502,6 +502,9 @@ void DuckDBPyRelation::WriteCsvDF(py::object df, const string &file, DuckDBPyCon
 // should this return a rel with the new view?
 unique_ptr<DuckDBPyRelation> DuckDBPyRelation::CreateView(const string &view_name, bool replace) {
 	rel->CreateView(view_name, replace);
+	// We need to pass ownership of any Python Object Dependencies to the connection
+	auto all_dependencies = rel->GetAllDependencies();
+	rel->context.GetContext()->external_dependencies[view_name] = move(all_dependencies);
```

Comment:
Rather than moving the dependencies here, should we not turn the dependencies into shared pointers, and keep them in both the relation and the view?

I imagine the move might break if we create a view, unregister the view and then try to use the relation again. Could you add a test for this?

Reply:
Oh yeah, that's a great point, good catch.